less_users/views.py

```diff
@@ -40,104 +48,231 @@ class ProfileView(LoginRequiredMixin, View):
+    except ValueError:
+        messages.warning(request, "This challenge does not exist")
+    else:
+        if Challenge.objects.filter(pk=challenge_id).exists():
+            points = Challenge.objects.get(pk=challenge_id).points
+            # check also not active user_challenges and their logs for frequency!
+            if UserChallenge.objects.filter(
+                user_id=request.user, challenge_id=challenge_id
+            ).exists():
+                user_challenge = UserChallenge.objects.get(
+                    user_id=request.user,
+                    challenge_id=challenge_id,
+                    is_active=True,
+                    is_deleted=False,
+                )
+                if user_challenge:
+                    user_challenge.check_if_active
+                    if user_challenge.is_active:
+                        """proceed user getting points for user challenge; it is up to date"""
+                        new_points = user_challenge.get_points
+                        if new_points == points:
+                            Log.objects.create(
+                                user_challenge_id=user_challenge.id, points=points
+                            )
+                            my_profile = request.user.profile
+                            my_profile.points += points
+                            my_profile.save()
+                            messages.success(
+                                request,
+                                f"You have just got {new_points} points for challenge: {user_challenge.challenge}!",
+                            )
+                        else:
+                            messages.warning(
+                                request,
+                                f"You cannot get new points for this challenge yet: {user_challenge.challenge}. "
+                                f"You have already done it within last {user_challenge.challenge.frequency} days",
+                            )
+                    else:
+                        # try to check previous logs, even for other user_challenges, but for the same user and challenge
+                        # and even if they are not active or are deleted, it counts points!
+                        prev_logs = Log.objects.filter(
+                            user_challenge__user_id=request.user
+                        ).filter(user_challenge__challenge_id=challenge_id)
+                        if prev_logs:
+                            last_log = prev_logs.latest("date").date.date()
+                            if (
+                                last_log
+                                + timedelta(
+                                    days=Challenge.objects.get(
+                                        pk=challenge_id
+                                    ).frequency
+                                )
+                                > date.today()
+                            ):
```

Comment:
` last_log + timedelta( days=Challenge.objects.get(pk=challenge_id ).frequency)> date.today()`
nit: z tego możesz zrobić osobną metodę, żeby zwiększyć czytelność.


A teraz nie nit: wyciągasz już wcześniej challenge, nie duplikuj requestów :)

Reply:
Zmieniłam to w kodzie. Zrobiłam przypisanie:
`challenge = Challenge.objects.get(pk=challenge_id)`
a potem odwołuję się do wartości:
`challenge.points` i `challenge.frequency`
O to chodzi?

Reply:
Chyba tak, ale dodałaś komentarz zanim wrzuciłaś zmiany, więc nie mam pewności